### DISCUSSION TITLE
How to install Paho pywheel in JupyterLite (browser)

### DISCUSSION THREAD

**Discussion by gpeyton** [pyodide/pyodide#4194] [General]
Hi, 

I'm using JupyterLite, which uses PyDiode to use packages in a browser. I'm trying to load the PahoMQTT wheel but not sure how to do it? 

Here's the [wheel link](https://www.piwheels.org/project/paho-mqtt/). 

Here's the [example notebook](https://discoverling.github.io/morse-decoder/lab/index.html) I'm running (see morse_decoder.ipynb) 

Here's my attempt to load Paho, which is incorrect:

```
import pyodide_kernel
import pyodide_js
await pyodide_js.loadPackage('numpy')
await pyodide_js.loadPackage('scikit-learn')
await pyodide_js.loadPackage('https://www.piwheels.org/simple/paho-mqtt/paho_mqtt-1.6.1-py3-none-any.whl#sha256=c44b3dd1b298894c44e3e842f7b0ca3ebe01f628a6f229c881b2324613d7bba7')
pyodide_kernel.__version__    
```

**Comment by rth** [comment]
Try,
```py
import micropip

await micropip.install(['numpy', 'scikit-learn', 'https://www.piwheels.org/simple/paho-mqtt/paho_mqtt-1.6.1-py3-none-any.whl#sha256=c44b3dd1b298894c44e3e842f7b0ca3ebe01f628a6f229c881b2324613d7bba7'])
```
that will install it but I doubt it would be usable, as MQTT works over TCP/IP sockets which are not directly accessible in a browser. It apparently can work with WebSockets, but someone would need to write a Python package with that implementation.

**Reply by gpeyton** [reply]
Yes, that installs, but execution times out on attempting to connect the socket.
